File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = size(prices), profit = 0, minBuy = prices[0];
        for(int &price : prices) {
            int sellingPrice = price - minBuy;
            profit = max(profit, sellingPrice);
            minBuy = min(minBuy, price);
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int minBuy = prices[0], ans = 0;
        for(int price : prices) {
            ans = Math.max(ans, price - minBuy);
            minBuy = Math.min(minBuy, price);
        }
        return ans;
    }
}